internal/io/spinner.go

```diff
@@ -1,22 +1,44 @@
 package io
 
 import (
+	"fmt"
 	"os"
+	"time"
 
-	"github.com/charmbracelet/huh/spinner"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
 	"github.com/mattn/go-isatty"
 )
 
-func SpinWhile(name string, action func()) error {
-	if !isatty.IsTerminal(os.Stdout.Fd()) {
-		// No TTY available, just run the action without spinner
+func SpinWhile(f *factory.Factory, name string, action func()) error {
+	// If quiet mode is on or not a terminal, just run the action
+	if f.Quiet || !isatty.IsTerminal(os.Stderr.Fd()) {
```

Comment:
Should we keep using `stdout`? It looks like other places in the codebase rely on `stdout`.